src/http/lib/pubsub.ts

```diff
@@ -23,30 +23,34 @@ export default class DaprPubSub {
     return res.status;
   }
 
-  async subscribe(pubSubName: string, topic: string, cb: TypeDaprPubSub) {
+  async subscribe(subs: TypeElementOfDaprPubSub[]) {
     const server = await WebServerSingleton.getServer();
+    if (subs.length < 1) return;
+    
+    subs.forEach((sub: TypeElementOfDaprPubSub, i) => {
+     
+      server.post(`/${sub.route}`, async (req, res) => {
+        console.log(`[Dapr API][PubSub][route-${sub.topic}] Handling incoming message`);
+
+        // Process our callback
+        await sub.cb(req, res);
+
+        // Let Dapr know that the message was processed correctly
+        // console.log(`[Dapr API][PubSub][route-${topic}] Ack'ing the message`);
+        return res.send({ success: true });
+      });
+    });
 
     server.get('/dapr/subscribe', (req, res) => {
-      console.log(`[Dapr API][PubSub][route-${topic}] Registering route for queue ${pubSubName}`);
-
-      res.send([
-        {
-          pubsubname: pubSubName,
-          topic,
-          route: `route-${pubSubName}-${topic}`,
-        },
-      ]);
+      console.log(`[Dapr API][PubSub] Registering`);
+      let subsCopy : TypeElementOfDaprPubSub[] = JSON.parse(JSON.stringify(subs));
+      subsCopy = subsCopy.map((row:TypeElementOfDaprPubSub)=>{
+        row.cb = null;
```

Comment:
I don't like this implementation tbh, can you think of a different way to do this?

Reply:
it is just for removing cb from data, there is no problem at all sending cb to Dapr.

Reply:
Yes, but this should be simpler I think? e.g. https://stackoverflow.com/a/53443378

Reply:
object assign not work in this case bkz it is shallow copy and we need deep copy.